### DISCUSSION TITLE
微软的Edge和IE浏览器中无法正常加载Waline

### DISCUSSION THREAD

**Discussion by elkan1788** [walinejs/waline#536] [Q&A] [answered]
在测试不同浏览器访问`waline`的效果时，发现在微软的Edge和IE浏览器中都无法正常加载`waline`，报错如下：
![image](https://user-images.githubusercontent.com/1217096/132595057-c0df509b-dea0-45d7-af16-879a0294cc02.png)

然后尝试了访问`waline`官网，结果还是一样无法正常显示页面，如下：
![image](https://user-images.githubusercontent.com/1217096/132595508-8f1c3100-2346-4186-94ba-ef2646acc461.png)

但是原来的Valine版本是可以正常的在Edge和IE浏览器中显示的，请问这是要特定的做些什么参数设置吗？

**Comment by lizheming** [comment, marked as answer]
Waline 兼容 Valine 指的是数据层面的，客户端这块因为我们使用了 Vue 3 开发，所以是不兼容 IE 和老版 Edge 的。你可以试试升级到新版 Edge 看看。
如果你对兼容性有要求的话可以看看 README 中列举的其它的客户端 https://minivaline.js.org/ 和 https://github.com/asforest/AprilComment 看一下

**Reply by elkan1788** [reply]
好的，谢谢你的提示，后续我再看看，谢谢。

**Comment by Mister-Hope** [comment]
可以这么说，这两款浏览器的市场份额已经几乎可以忽略了，作为一个主要被应用在博客上的、面向普通人的一个开源项目，我觉得drop掉历史版本的浏览器是很正常的。毕竟此项目的主要使用者也不是企业、政府等场景，需要保持强的兼容性。

换言之，从我们的角度，大部分用户需要体积和性能，兼容性次之（不过我们也的确在最近会聚焦提升一部分我们的兼容性范围）😁

但是ie和老板edge显然不在我们的考虑范围内。如果你的确有需求，可以考虑第三方客户端。

**Reply by elkan1788** [reply]
嗯，看来是自己要求想的多啦。确实老版本的浏览器兼容性对开发者是很痛苦的，特别是以前ie6的年代。
